services/openvpn/session/client_map.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package session
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/mysteriumnetwork/node/session"
+)
+
+type clientMap struct {
+	sessionManager   session.Manager
+	sessionClientIDs map[session.SessionID]int
+	sessionMapLock   sync.Mutex
+}
+
+// FindClientSession returns OpenVPN session instance by given session id
+func (cm *clientMap) FindClientSession(clientID int, id session.SessionID) (session.Session, bool, error) {
+	sessionInstance, foundSession := cm.sessionManager.FindSession(id)
+
+	if !foundSession {
+		return session.Session{}, false, errors.New("no underlying session exists, possible break-in attempt")
+	}
+
+	sessionClientID, clientIDFound := cm.sessionClientIDs[id]
```

Comment:
Decide on either clientIDFound or foundClientID, not both.

Reply:
Renamed